### PR TITLE
Numpy and actions

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- #22 numpy data accessor
- access data column by index if no column names
- #48 lambda in actions
- all actions are now named on_*

### CODE REVIEW

File: taipy/gui/data/pandas_data_accessor.py [hunk 1]
```diff
@@ -45,10 +48,10 @@ def __build_transferred_cols(
         self, gui: Gui, payload_cols: t.Any, data: pd.DataFrame, styles: t.Optional[t.Dict[str, str]] = None
     ) -> pd.DataFrame:
         if isinstance(payload_cols, list) and len(payload_cols):
-            col_types = data.dtypes[data.dtypes.index.isin(payload_cols)]
+            col_types = data.dtypes[data.dtypes.index.astype("string").isin(payload_cols)]
```
Comment (FabienLelaquais) [inline taipy/gui/data/pandas_data_accessor.py]: The (numpy) doc says that we should use 'astype(str)'. I wonder if that is not a bit more efficient and less error-prone...

File: taipy/gui/renderers/builder.py [hunk 2]
```diff
@@ -56,16 +56,17 @@ def __init__(
...
-            if not isinstance(properties_dict, _MapDict):
+            (prop_dict, prop_hash) = self.__parse_attribute_value(self.__attributes["properties"])
+            if prop_hash is None:
+                self.__gui.bind_var(prop_dict)
+                prop_dict = getscopeattr(self.__gui, prop_dict)
+            if not isinstance(prop_dict, _MapDict):
                 raise Exception(
-                    f"Can't find properties configuration dictionary {properties_dict_name}!"
+                    f"Can't find properties configuration dictionary {self.__attributes['properties'] if prop_hash is None else prop_hash}!"
                     f" Please review your GUI templates!"
```
Comment (FabienLelaquais) [inline taipy/gui/renderers/builder.py]: We don't want this line (and the '!' in the line above)

**Review by FabienLelaquais — APPROVED** [review]
Better, and better... well done!

**Review by dinhlongviolin1 — APPROVED** [review]
You are the best!!!! 👍 👍 💯 🥇

File: taipy/gui/data/pandas_data_accessor.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/data/pandas_data_accessor.py]: I'll try it

File: taipy/gui/renderers/builder.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply taipy/gui/renderers/builder.py]: What do you want then ?

Comment (FredLL-Avaiga) [reply taipy/gui/renderers/builder.py]: Changed this to a warning and rephrased

**Pull request merged**